connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerSourceTaskTest.java

```diff
@@ -706,95 +673,77 @@ public void testSourceTaskIgnoresProducerException() throws Exception {
         // and no ConnectException will be thrown
         SourceRecord record1 = new SourceRecord(PARTITION, OFFSET, TOPIC, 1, KEY_SCHEMA, KEY, RECORD_SCHEMA, RECORD);
         SourceRecord record2 = new SourceRecord(PARTITION, offset2, TOPIC, 2, KEY_SCHEMA, KEY, RECORD_SCHEMA, RECORD);
-        expectOffsetFlush(true);
-        expectSendRecordOnce();
-        expectSendRecordProducerCallbackFail();
-        sourceTask.commitRecord(EasyMock.anyObject(SourceRecord.class), EasyMock.isNull());
 
-        //As of KAFKA-14079 all offsets should be committed, even for failed records (if ignored)
-        //Only the last offset will be passed to the method as everything up to that point is committed
-        //Before KAFKA-14079 offset 12 would have been passed and not 13 as it would have been unacked
-        offsetWriter.offset(PARTITION, offset2);
-        PowerMock.expectLastCall();
+        expectOffsetFlush();
+        expectPreliminaryCalls();
 
-        PowerMock.replayAll();
+        when(producer.send(any(ProducerRecord.class), any(Callback.class))).thenAnswer(
+                producerSendAnswer(true)
+        ).thenAnswer(producerSendAnswer(false));
```

Comment:
Nit: A small whitespace change can help readability here.

```suggestion
        when(producer.send(any(ProducerRecord.class), any(Callback.class)))
                .thenAnswer(producerSendAnswer(true))
                .thenAnswer(producerSendAnswer(false));
```